File: 0003-longest-substring-without-repeating-characters/0003-longest-substring-without-repeating-characters.py
class Solution:
    def lengthOfLongestSubstring(self, s: str) -> int:
        start = 0
        # end =1
        sub =1
        seen ={}
        if (len(s)==0):
            return 0;
        for end in range(len(s)):
            if s[end] not in seen:
                sub = max(sub, end-start+1)

            else:
                if seen[s[end]]< start:
                    sub = max(sub, end-start+1)
                else:
                    start= seen[s[end]]+1
                
            seen[s[end]]= end
        


        

        return sub
            
            

        

File: 0003-longest-substring-without-repeating-characters/0003-longest-substring-without-repeating-characters.cpp
class Solution {
public:
    int lengthOfLongestSubstring(string s) {
        map<char, int> mp;
        int len=0, maxL=0;
        int j=0;
        for(int i=0;i<s.length();i++){
            mp[s[i]]++;
            len++;
            while(mp[s[i]]>1){
                mp[s[j]]--;
                j++;
                len--;
            }

            maxL = max(maxL, len);
        }

        return maxL;
    }
};